cmd/kubeadm-nspawn/kubeadm-nspawn.go

```diff
@@ -98,7 +129,16 @@ func newUpCommand() *cobra.Command {
 }
 
 func runInit(cmd *cobra.Command, args []string) {
-	if k8srelease == "" {
+	if !checkK8sStableRelease(k8srelease) {
+		log.Printf("WARNING: K8s with version <%s is not compatible with kubeadm-nspawn.",
+			k8sStableVersion)
+		log.Printf("It's highly recommended to upgrade K8s to 1.7 or newer.")
+		// Print a kind warning, and continue to run.
+		// It should still allow kubeadm to run with Kubernetes <1.7,
+		// to be able to allow end users to flexibly handle various cases.
```

Comment:
I would factor this out into a function which can be used by both `runInit` and `runUp`